openapi-validation-core/src/main/java/com/getyourguide/openapi/validation/core/OpenApiRequestValidator.java

```diff
@@ -32,11 +34,10 @@ public OpenApiRequestValidator(
         ThreadPoolExecutor threadPoolExecutor,
         ValidationReportHandler validationReportHandler,
         MetricsReporter metricsReporter,
-        String specificationFilePath,
-        ValidatorConfiguration configuration
+        OpenApiInteractionValidatorWrapper validator
     ) {
         this.threadPoolExecutor = threadPoolExecutor;
-        this.validator = new OpenApiInteractionValidatorFactory().build(specificationFilePath, configuration);
+        this.validator = validator;
```

Comment:
ℹ️  To mock the validator this needs to be injected. So I improved the architecture here (should have done this from the beginning).